pokemongo_bot/tests/bot_test.py

```diff
@@ -0,0 +1,574 @@
+        log_fn.assert_any_call('-- Level: 14', prefix='#')
+        log_fn.assert_any_call('-- Experience: 15', prefix='#')
+        log_fn.assert_any_call('-- Experience until next level: 985', prefix='#')
+        log_fn.assert_any_call('-- Pokemon captured: 17', prefix='#')
+        log_fn.assert_any_call('-- Pokestops visited: 18', prefix='#')
+
+        bot.event_manager.fire_with_context.assert_has_calls([
+            call('bot_initialized', bot),
+            call('pokemon_bag_full', bot),
+            call('item_bag_full', bot)
+        ])
+
+        # Check logging
+        assert logging.getLogger('requests').level == logging.DEBUG
+        assert logging.getLogger('pgoapi').level == logging.DEBUG
+        assert logging.getLogger('rpc_api').level == logging.DEBUG
+
+        # Check Plugins
+        self._assert_plugins_loaded(bot.plugin_manager, ['test_plugin'])
+
+        # Check events
+        assert bot.event_manager.fire_with_context.call_count == 3
+
+    @patch('pokemongo_bot.event_manager.manager', new_callable=EventManager)
+    @patch('pokemongo_bot.logger.log', return_value=None)
+    def test_start_login_success_location_cache_not_exists(self, log_fn, event_manager):
+        bot = self._create_generic_bot({
+            'debug': True,
+            'test': False,
+            "location_cache": True,
+            "print_events": True,
+            'exclude_plugins': ['another_test_plugin'],
+            'load_library': 'libencrypt.so',
+            'auth_service': 'ptc',
+            'username': 'test_account',
+            'password': 'pa55w0rd',
+            'location': None,
+            'gmapkey': 'test_gmaps_key',
+            'initial_transfer': True,
+            'recycle_items': True
+        })
+
+        if os.path.isfile('data/last-location-test_account.json'):
+            os.unlink('data/last-location-test_account.json')
+
+        log_fn.return_value = None
+
+        event_manager.fire_with_context = Mock()
+        bot.event_manager = event_manager
+
+        with pytest.raises(SystemExit):
+            bot.start()
+
+    @patch('pokemongo_bot.event_manager.manager', new_callable=EventManager)
+    @patch('pokemongo_bot.logger.log', return_value=None)
+    def test_start_login_success_location_cache(self, log_fn, event_manager):
+        bot = self._create_generic_bot({
+            'debug': True,
+            'test': False,
+            "print_events": True,
+            'exclude_plugins': ['another_test_plugin'],
+            'load_library': 'libencrypt.so',
+            'auth_service': 'ptc',
+            'username': 'test_account',
+            'password': 'pa55w0rd',
+            'location': '0,0',
+            'location_cache': True,
+            'gmapkey': 'test_gmaps_key',
+            'initial_transfer': True,
+            'recycle_items': True
+        })
+
+        if os.path.isfile('data/last-location-test_account.json'):
+            os.unlink('data/last-location-test_account.json')
```

Comment:
These files/accounts REALLY need to be named differently. The reason is because attempting to multithread the test runner (which is how it's set up on Travis and AppVeyor at the moment) can result in race conditions if two tests that access this file are run at the same time.